src/components/Sep8Send/Sep8Review.tsx

```diff
@@ -0,0 +1,132 @@
+import { useEffect, useState } from "react";
+import { useDispatch } from "react-redux";
+import { Transaction, TransactionBuilder } from "stellar-sdk";
+import { Button, Heading2, Loader } from "@stellar/design-system";
+import { Json } from "components/Json";
+import { Modal } from "components/Modal";
+import { Toggle } from "components/Toggle";
+import { fetchAccountAction } from "ducks/account";
+import { sep8SubmitRevisedTransactionAction } from "ducks/sep8Send";
+import { getNetworkConfig } from "helpers/getNetworkConfig";
+import { useRedux } from "hooks/useRedux";
+import { ActionStatus } from "types/types.d";
+
+export const Sep8Review = ({ onClose }: { onClose: () => void }) => {
+  const { account, sep8Send, settings } = useRedux(
+    "account",
+    "sep8Send",
+    "settings",
+  );
+  const [submittedTx, setSubmittedTx] = useState<Transaction | undefined>();
+  const [revisedTx, setRevisedTx] = useState<Transaction | undefined>();
+  const [isApproved, setIsApproved] = useState(false);
+  const dispatch = useDispatch();
+
+  // user interaction handlers
+  const handleSubmitPayment = () => {
+    if (sep8Send.data.revisedTransaction.revisedTxXdr && isApproved) {
+      dispatch(sep8SubmitRevisedTransactionAction());
+    }
+  };
+
+  // use effect: complete action, close modal and refresh account balances
+  useEffect(() => {
+    if (sep8Send.status === ActionStatus.SUCCESS && account.data?.id) {
+      dispatch(
+        fetchAccountAction({
+          publicKey: account.data.id,
+          secretKey: account.secretKey,
+        }),
+      );
+      onClose();
+    }
+  }, [account.data?.id, account.secretKey, sep8Send.status, dispatch, onClose]);
+
+  // use effect: parse transaction XDRs
+  const { revisedTxXdr, submittedTxXdr } = sep8Send.data.revisedTransaction;
```

Comment:
We could move this up and use destructured value here `if (sep8Send.data.revisedTransaction.revisedTxXdr && isApproved) {` (line 27)

Reply:
Good point, done in https://github.com/stellar/stellar-demo-wallet/pull/157/commits/28fd0c44b5fd5428f65295f0a462e77bbb61596c.